Libs/MRML/Core/vtkMRMLModelDisplayNode.cxx

```diff
@@ -186,7 +186,7 @@ void vtkMRMLModelDisplayNode::SetThresholdRange(double min, double max)
 {
   vtkMTimeType mtime = this->ThresholdFilter->GetMTime();
   this->ThresholdFilter->ThresholdBetween(min, max);
-  if (mtime >= this->ThresholdFilter->GetMTime())
+  if (mtime >= this->ThresholdFilter->GetMTime() || !this->GetInputMesh())
```

Comment:
The node content (threshold value) is still modified, so we should not return - regardless of having a valid input mesh or not.
We should just avoid calling this->ThresholdFilter->Update(). By the way this->ThresholdFilter->Update() call should not be necessary, because VTK takes care of running the filters when data is requested (and it can actually slow things down that we force an update instead of letting the renderer to compute data when needed).
Could you try removing this->ThresholdFilter->Update() and see if everything works well? The code could be simplified to this then:
<pre>
vtkMTimeType mtime = this->ThresholdFilter->GetMTime();
this->ThresholdFilter->ThresholdBetween(min, max);
if (this->ThresholdFilter->GetMTime() > mtime) 
  {
  this->Modified();
  }
</pre>